From 3beb9f5938175ed8c6fd38e3fccd3606ce5f5c99 Mon Sep 17 00:00:00 2001
From: pdiopX <piotrx.diop@intel.com>
Date: Fri, 29 Mar 2013 16:32:38 +0100
Subject: Use built-in mic whenever IHF is forced as output device in call

BZ: 95092
[CLEAN]

When in communication with a connected headset/bluetooth device, the user
was forced to use that device's embedded mic even when he wanted to have
the output on speaker.

This fix allows selecting the built-in mic whenever the user wants to
force the speaker for output when in communication.

Category: device enablement
Domain: AudioComms-Common
Origin: internal
Upstream-Candidate: yes
Change-Id: Iaa85171e505a7721e784e99f7c5b9694cb994422
Orig-Change-Id: I435b1572849f492c7f20920d42b9856933e39a0d
Signed-off-by: pdiopX <piotrx.diop@intel.com>
---
 audio/AudioPolicyManagerBase.cpp |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index f590bb5..425b231 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -2785,11 +2785,16 @@ audio_devices_t AudioPolicyManagerBase::getDeviceForInputSource(int inputSource)
       }
       // FALL THROUGH
 
+    case AUDIO_SOURCE_VOICE_COMMUNICATION:
+        if (mForceUse[AudioSystem::FOR_COMMUNICATION] == AudioSystem::FORCE_SPEAKER &&
+            mAvailableInputDevices & AUDIO_DEVICE_IN_BUILTIN_MIC) {
+            device = AUDIO_DEVICE_IN_BUILTIN_MIC;
+            break;
+        }
     case AUDIO_SOURCE_DEFAULT:
     case AUDIO_SOURCE_MIC:
     case AUDIO_SOURCE_VOICE_RECOGNITION:
     case AUDIO_SOURCE_HOTWORD:
-    case AUDIO_SOURCE_VOICE_COMMUNICATION:
         if (mForceUse[AudioSystem::FOR_RECORD] == AudioSystem::FORCE_BT_SCO &&
             mAvailableInputDevices & AUDIO_DEVICE_IN_BLUETOOTH_SCO_HEADSET) {
             device = AUDIO_DEVICE_IN_BLUETOOTH_SCO_HEADSET;
-- 
1.7.4.1

